File: examen_1/src/ejercicio_3_2.f90
program es_primo

    implicit none
    integer :: a, resultado

    print *, "Digita un numero: "
    read *, a

    call is_p(a, resultado)

    if (resultado == 1) then
        print *, "true"
    else 
        print *, "false"
    end if

end program es_primo

subroutine is_p(a, resultado)

    integer :: a, i, resultado

    if (a >= 1 .AND. a<=3) then 
        resultado = 1
        return 
    end if

    do i=2, a - 1
        if (mod(a, i) == 0) then
            resultado = 0
            return
        end if 
    end do

    resultado = 1

end subroutine is_p

File: examen_1/src/ejercicio_3_2.c
#include <stdio.h>

int is_p(int a);

int main(int argc, char **argv) {

    int number = 0;

    printf("Digita un numero: ");
    scanf("%d", &number);

    printf("Es numero primo:  %s\n", is_p(number) ? "true" : "false" );

    return 0;
}

int is_p(int a){
    if ( a >=1 && a <= 3 ) return 1;

    for (int i=2; i<a;i++){
        if ( a % i == 0) return 0;
    }
    return 1;
}